packages/salesforcedx-utils-vscode/package.json

```diff
@@ -11,7 +11,8 @@
   ],
   "main": "out/src",
   "dependencies": {
-    "@salesforce/core": "^3.31.18",
+    "@salesforce/source-tracking": "2.2.24",
+    "@salesforce/core": "3.34.6",
```

Comment:
Add source-tracking to the utils-vscode pkg and upgrade core to align version with core pkg

Reply:
I think I asked this is in another PR, but I'm losing track haha. Does core need to be consistent across our packages?

Reply:
It seems to me that even though the package.json entries are not all exactly the same across the packages, we end up with only two versions of core installed 3.34.6 and 2.37.1 (for Functions):
<img width="500" alt="2023-04-14_16-26-47" src="https://user-images.githubusercontent.com/46458081/232170242-2eac3609-b544-4d6b-aeff-4dd1be672313.png">


<img width="500" alt="2023-04-14_16-26-25" src="https://user-images.githubusercontent.com/46458081/232170323-38b81a31-653d-4acc-a20d-0b5a1dc97022.png">

Reply:
A new version of core was released, which resulted in npm i installing 3 versions of @salesforce/core, and then hitting compilation errors.  In #4808 all versions of core dependencies in vsce packages were aligned on v 3.34.7 to avoid the compilation errors.